src/OpenApi/src/Extensions/OpenApiEndpointRouteBuilderExtensions.cs

```diff
@@ -50,7 +50,7 @@ public static IEndpointConventionBuilder MapOpenApi(this IEndpointRouteBuilder e
                 else
                 {
                     var document = await documentService.GetOpenApiDocumentAsync(context.RequestServices, context.RequestAborted);
-                    var documentOptions = options.Get(documentName);
+                    var documentOptions = options.Get(lowercasedDocumentName);
```

Comment:
Is there a test that can updated that would fail before your change and pass afterwards?

Reply:
I can't really find a way to write a test for this. Looking at the source code, if the options can't find a value, it will add it instead. As this is an integration test, I don't have access to the instances and can't mock or verify method calls.

Reply:
Would an HTTP request to a test server hosting an OpenAPI endpoint find the bug you're fixing?

Reply:
Never mind, I found a way! Working on it now..

Reply:
@martincostello I've added the test. Feel free to review. For now, I'll resolve this conversation as the test has been added.